TTS/encoder/models/lstm.py

```diff
@@ -182,8 +182,18 @@ def batch_compute_embedding(self, x, seq_lens, num_frames=160, overlap=0.5):
     def load_checkpoint(self, config: dict, checkpoint_path: str, eval: bool = False, use_cuda: bool = False):
         state = load_fsspec(checkpoint_path, map_location=torch.device("cpu"))
         self.load_state_dict(state["model"])
+        # load the criterion for emotion classification
```

Comment:
I would separate this into different loss classes as we do it for the TTS models

Reply:
The plan is to load this loss just for use in inference time. It is not loaded in --restore_path. For this reason, I added all these ifs. "config.map_classid_to_classname" will be different of None just during the inference. And the only Loss that we need to load is SoftmaxAngleProtoLoss because it has an nn.Dense layer.

Reply:
I create a EncoderBase Class that includes common encoder specific functions. In addition, I add a get_criterion to return the right criterion based on the config.loss.